### DISCUSSION TITLE
Help troubleshooting custom snippets not showing up

### DISCUSSION THREAD

**Discussion by nadiamoe** [saghen/blink.cmp#654] [Q&A] [answered]
Hey folks,

New blink.cmp user here trying it out! So far I'm really liking the responsiveness and the fuzzyness!

I'm having a bit of trouble troubleshooting custom snippets though. I have them defined what I understand is the default path:
https://github.com/Saghen/blink.cmp/blob/46eb5f3fd3dba0a3bb3d026fb2b7462f4a0656b7/lua/blink/cmp/sources/snippets/registry.lua#L17

```
nadia@Nadiarch
11:21:30 ~ $> l ~/.config/nvim/snippets
total 29K
drwxr-xr-x 2 nadia users    4 Jul 18 13:00 .
drwxr-xr-x 4 nadia users   13 Nov  3 21:30 ..
-rw-r--r-- 1 nadia users 1.5K Jul 18 13:00 go.snippets
-rw-r--r-- 1 nadia users 9.8K Nov  3  2023 yaml.snippets
```

They are in this format, which I understand is one of the standards:
```
11:22:00 ~ $> head ~/.config/nvim/snippets/go.snippets
snippet fn "anonymous func"
	func(${1}) {
		${0}
	}

snippet fnc "anonymous func call"
	func(${1}) {
		${0}
	}()


```

However, they are not showing up in the completion menu:
![image](https://github.com/user-attachments/assets/ff898e19-29ae-407d-afb9-7bd15dc0eda6)

My config snippet is:
```lua
  {
    "saghen/blink.cmp",
    ---@module 'blink.cmp'
    ---@type blink.cmp.Config
    opts = {
      keymap = {
        preset = "super-tab",
      },
      completion = {
        ghost_text = {
          enabled = false,
        },
      },
    },
  },
```

Which should get merged with the LazyVim config here: https://github.com/LazyVim/LazyVim/blob/main/lua/lazyvim/plugins/extras/coding/blink.lua

I'd love to see some pointers on how to debug this further. How can I know what is not working here?
Thanks!

**Comment by Quitlox** [comment, marked as answer]
From the [documentation](https://cmp.saghen.dev/configuration/snippets.html): "Currently, only VSCode style snippets are supported, but you may look into [Luasnip](https://github.com/L3MON4D3/LuaSnip) if you'd like more advanced functionality."

The snippet you're providing is not a VSCode style snippet: https://code.visualstudio.com/docs/editor/userdefinedsnippets

**Reply by nadiamoe** [reply]
Gotcha, thank you!

**Reply by nadiamoe** [reply]
In case someone else stumbles upon this, I have made this tiny thing that takes snipmate snippets and converts them to JSON: https://github.com/nadiamoe/snipmate-to-json

It is not well tested and there may be dragons, but worked for me.

**Reply by saghen** [reply]
Very cool! Added it to the docs https://main.cmp.saghen.dev/configuration/snippets.html#custom-snippets
